lib/cocoapods/target/build_settings.rb

```diff
@@ -559,12 +559,16 @@ def initialize(target, non_library_spec = nil)
 
           frameworks = []
           frameworks.concat consumer_frameworks
-          if library_xcconfig? && (target.should_build? && target.build_as_dynamic?)
-            frameworks.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
-          end
-          if non_library_xcconfig?
+          if library_xcconfig?
+            # We know that this library target is being built dynamically based
+            # on the guard above, so include any vendored static frameworks.
+            frameworks.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') } if target.should_build?
```

Comment:
does this mean that a vendored static framework will be linked _twice_? Once for this dynamic framework and once on the app target itself?

Reply:
This line is just moved around, but no - since this is a dynamic library target, when it gets compiled it needs to include any vendored static frameworks, and then the whole dynamic library will get linked into it's parent (either app or another library).

Reply:
If I recall correctly CocoaPods will link that vendored static framework to the app target as well not just the dynamic library target.

From its early days CocoaPods had a very (kinda annoying) naive approach to just link everything to the app level target.

Reply:
I can verify my own question this week.

Reply:
I guess you are right this was already happening before. I will still verify though